File: acropolis-ui/src/models/resource.ts
export interface Resource {
  id: string;
  storageLocation: string;
  createdTimestamp: string;
  views?: number
}

export interface ResourceModel {
  id: string;
  title: string | null;
  description: string;
  url: string;
  source: string;
  resources: Resource[]
  type: 'page' | 'video';
}

export interface Page extends ResourceModel {
  type: 'page';
}

export interface Video extends ResourceModel {
  type: 'video';
}

export interface PendingResource {
  id: string;
  url: string;
  requestedTimestamp: string;
  currentState: string;
  error?: PendingResourceError
}

export interface PendingResourceError {
  errorMessage?: string;
  errorTimestamp?: string;
}

File: Acropolis/Acropolis.Domain/Resource.cs
namespace Acropolis.Domain;

public class Resource
{
    public Resource(Guid id, string storageLocation, DateTimeOffset createdTimestamp)
    {
        Id = id;
        StorageLocation = storageLocation;
        CreatedTimestamp = createdTimestamp;
    }
    
    public Guid Id { get; private set; }

    public string StorageLocation { get; private set; }
    public DateTimeOffset CreatedTimestamp { get; private set; }
    public int Views { get; private set; }
    
    public void IncrementViews() => Views++;
    public void ResetViews() => Views = 0;
}